nxdrive/gui/api.py

```diff
@@ -386,6 +387,20 @@ def generate_report(self) -> str:
             log.exception("Report error")
             return "[ERROR] " + str(e)
 
+    @pyqtSlot(str, str, result=bool)
+    def generate_csv(self, session_id: str, engine_uid: str) -> bool:
+        """
+        Generate a CSV file from the *session_id*.
+        """
+        try:
+            engine = self._manager.engines.get(engine_uid)
+            if not engine:
+                return False
```

Comment:
You should move that logic outside the try/except block.
```suggestion
            engine = self._manager.engines.get(engine_uid)
            if not engine:
                return False
```